Library/Homebrew/requirements/java_requirement.rb

```diff
@@ -45,16 +29,86 @@ def inspect
 
   def display_s
     if @version
-      if @version[-1] == "+"
-        op = ">="
-        version = @version[0, @version.length-1]
-      else
+      if exact_version?
         op = "="
-        version = @version
+      else
+        op = ">="
       end
-      "#{name} #{op} #{version}"
+      "#{name} #{op} #{version_without_plus}"
     else
       name
     end
   end
+
+  private
+
+  def version_without_plus
+    if exact_version?
+      @version
+    else
+      @version[0, @version.length-1]
+    end
+  end
+
+  def exact_version?
+    @version && @version.to_s.chars.last != "+"
+  end
+
+  def setup_java
+    java = preferred_java
+    return unless java
+    @java = java
+    @java_home = java.parent.parent
+  end
+
+  def possible_javas
+    javas = []
+    javas << Pathname.new(ENV["JAVA_HOME"])/"bin/java" if ENV["JAVA_HOME"]
+    jdk = begin
+      Formula["jdk"]
+    rescue FormulaUnavailableError
+      nil
+    end
+    javas << jdk.bin/"java" if jdk && jdk.installed?
+    javas << which("java")
+    javas
+  end
+
+  def preferred_java
+    possible_javas.detect do |java|
+      next false unless java && java.executable?
+      next true unless @version
+      next true if satisfies_version(java)
+    end
+  end
+
+  def env_java_common
+    java_home = Pathname.new(@java_home)
+    ENV["JAVA_HOME"] = java_home
+    ENV.prepend_path "PATH", java_home/"bin"
+  end
+
+  def env_oracle_jdk
+    java_home = Pathname.new(@java_home)
```

Comment:
Suggestion: add `return unless @java_home` and then `return unless (java_home/"include").exist?` and `true` as the last value. Then you can use this method chained to remove the need for checks in multiple places.